cache.c

```diff
@@ -467,12 +467,21 @@ struct track_info **cache_refresh(int *count, int force)
 {
 	struct track_info **tis = get_track_infos();
 	int i, n = total;
+	uint64_t last_break = time_in_ms();
 
 	for (i = 0; i < n; i++) {
 		unsigned int hash;
 		struct track_info *ti = tis[i];
 		struct stat st;
 		int rc = 0;
+		uint64_t now;
+
+		now = time_in_ms();
+		if (now - last_break > 100) {
+			cache_unlock();
```

Comment:
On linux at the very least pthread_mutex_unlock reduces to a single atomic operation if no other thread is waiting.